Multiple currencies

eddiepease opened the pull request:
Added the ability to donate in multiple different currencies. Specifically by:

- Adding functionality to convert between any two currencies, together with appropriate unit testing
- Adjusting front-end to select a major currency to donate in (sats, USD, EUR, AUD, JPY, GBP)

Also as part of the PR, I have removed unused dependencies in the files I have used / updated formatting to PEP8. Happy to put this as part of a separate PR if you'd prefer.

kristapsk commented:
> Also as part of the PR, I have removed unused dependencies in the files I have used / updated formatting to PEP8.

This duplicates #71, but let's see what gets merged first and then resolve conflicts. :)

kristapsk commented:
Two comments (just my opinion) by looking at code (haven't yet tested):

1. Currency list should be configurable, not every user will want to have all supported foreign fiat currencies. Some may even just want BTC / sats.
2. For Bitcoin I think both BTC and sats should be supported. People are mostly used to deal with BTC values when doing onchain payments and sats when doing LN payments.

kristapsk commented:
Also selected currency field (`fiat_currency`) should be added added to `payments` table, there is just `fiat_value` currently. See `payments/database.py`. Should not be added in `createt_database()`, but `migrate_database()`.

kristapsk commented:
Now that #71 is merged, also needs rebase.

eddiepease commented:
I agree with your comments about the currency list being configurable and that BTC and sats should be supported. I have just rebased to the new master and modified the payments table via migrate_database. I will tackle the currency list config shortly

kristapsk commented:
But it looks you did `git merge master`, better is to do `git rebase master` instead for less spam in git history. See https://github.com/bitcoin/bitcoin/blob/8abe79aedd0ba129e0fd3bcd971e8733d22fb3c4/CONTRIBUTING.md#squashing-commits about squashing commits and rebasing (or could just open a new PR against current master and close this one).

eddiepease commented:
Sounds good - will create a new PR after I've tackled the currency configuration work

eddiepease commented:
Creating a new PR as discussed above
